src/charm.py

```diff
@@ -30,6 +31,60 @@ def __init__(self, *args: typing.Any) -> None:
         super().__init__(*args)
         self.framework.observe(self.on.config_changed, self.reconciliation)
 
+    def _application_config(self) -> dict | None:
+        """Decode the value of the charm configuration application-config.
+
+        Returns:
+            The value of the charm configuration application-config.
+
+        Raises:
+            ReconciliationError: when application-config is invalid.
+        """
+        try:
+            config = self.model.config["application-config"]
+            if not config:
+                return None
+            application_config = json.loads(config)
+            if isinstance(application_config, dict):
+                return application_config
+            raise ReconciliationError(
+                new_status=BlockedStatus(
+                    "Invalid application-config value, expecting an object in JSON"
+                )
+            )
+        except json.JSONDecodeError as exc:
+            raise ReconciliationError(
+                new_status=BlockedStatus("Invalid application-config value, expecting JSON")
+            ) from exc
+
+    def _spring_boot_port(self) -> int:
+        """Get Spring Boot server port.
+
+        Returns:
+            Sprint Boot server port.
+        """
+        application_config = self._application_config()
+        if (
+            application_config
+            and "server" in application_config
+            and isinstance(application_config["server"], dict)
+            and application_config["server"]["port"]
+        ):
+            return application_config["server"]["port"]
+        return 8080
```

Comment:
Is it worth noting in the logs why `8080` was picked? E.g., that the configuration for the port was not found in the application configuration passed in by the user?

Reply:
Updated, thanks!